src/inference/src/dev/core_impl.cpp

```diff
@@ -779,6 +780,13 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
         cacheContent.blobId = ov::ModelCache::compute_hash(model, create_compile_config(plugin, parsed._config));
         cacheContent.model = std::const_pointer_cast<ov::Model>(model);
         std::unique_ptr<CacheGuardEntry> lock = cacheGuard.get_hash_lock(cacheContent.blobId);
+
+        const auto& rt_info = model->get_rt_info();
```

Comment:
See #29354, the logic of this part is changed.

Reply:
This logic still does not work if user didn't set `weights_path` hint. cacheContent.modelPath is empty on this way and weights path couldn't be compiled.